GitPrompt.ps1

```diff
@@ -145,13 +145,31 @@ function Write-GitStatus($status) {
     }
 }
 
-if(!(Test-Path Variable:Global:VcsPromptStatuses)) {
-    $Global:VcsPromptStatuses = @()
+$VcsPromptStatusesVariableName = "VcsPromptStatuses"
+
+if (-not (Test-Path Variable:Global:$VcsPromptStatusesVariableName)) {
+    New-Variable -Name $VcsPromptStatusesVariableName -Value @() -Scope Global
+}
+
+function Global:Write-VcsStatus {
+    Get-Variable -Name $VcsPromptStatusesVariableName -Scope Global -ValueOnly | foreach { 
+        try {
+            $_.Invoke()
+        }
+        catch {
+            Write-Error "A Write-VcsStatus scriptblock failed with the error: $($_.Exception.InnerException)"

```

Comment:
How about something like:
PS C:\> $Error
PS C:\> Write-Error -Message "test error" -ErrorAction SilentlyContinue
PS C:\> $Error
Write-Error -Message "test error" -ErrorAction SilentlyContinue : test error
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException

Then the normal Write-Error message isn't displayed but it is still in the error stream & error array so it can be debugged.

You are right as it executed in the prompt so shouldn't be intrusive, following that line of thought If there is an error and it is logged (silently), should doing any Write-Host be avoided as we don't know how it will affect the user's prompt? Or a message just saying [Error] as per your comment?
I am unsure if [error] was intended to be the error message or the actual word '[error]'.